sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java

```diff
@@ -88,6 +88,13 @@ public interface TableCatalog extends CatalogPlugin {
    */
   String PROP_OWNER = "owner";
 
+  /**
+   * A reserved property to specify the view text of a general table that represents
+   * a SQL view. The identifiers must be fully qualified in the view text to be
+   * context-independent, otherwise the behavior is undefined.
```

Comment:
I think the current view implementation which stores the original SQL text and a bunch of context is too convoluted to put into the public DS v2 API. It's better if the view text is context-independent. We are going to improve it in https://github.com/apache/spark/pull/51410

Before the improvement is done, we only allow to read DS v2 views that has context-independent SQL text.

Reply:
Don't we store the current catalog and namespace in the view metadata? Do we expect the connectors to make modify the view SQL text?

Reply:
So all identifiers in the view text will always include the catalog name as the first name part and the table name as the last name part? How hard will it be to modify the original SQL text? Will it cause any surprises to the users if the original and the persisted SQL text differ?